datadog_checks_base/datadog_checks/base/utils/discovery/cache.py

```diff
@@ -0,0 +1,27 @@
+# (C) Datadog, Inc. 2019-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import time
+
+
+class Cache:
+    def __init__(self, get_items_func, interval):
+        self._get_items_func = get_items_func
+        self._interval = interval
+        self._last_get_items_time = None
+        self._cached_items = []
+
+    def get_items(self):
+        if not self.__interval_configured() or self.__should_refresh_now():
+            self.__refresh()
+        return self._cached_items
+
+    def __interval_configured(self):
+        return self._interval is not None and self._interval > 0
+
+    def __should_refresh_now(self):
+        return self._last_get_items_time is None or (time.time() > self._last_get_items_time + self._interval)
+
+    def __refresh(self):
+        self._cached_items = self._get_items_func()
```

Comment:
I'm worried that in some cases this list could be big, depending on the service we are hitting and the number of projects they hold. Should we add a warning or something like that in this case (although I don't know yet what "big" means 😉)? Or do we just leave it to the user (in this case us) and they should make sure the return of their `_get_items_func` will not takes too much memory or handle that on their side?

Reply:
I agree with you, but I see it more related to performance tests or specific resource consumption of each integration.